tests/unit/api/main.py

```diff
@@ -1343,6 +1343,23 @@ def test_list_switches(self):
         ]
 
 
+class Test_show_switch:
+
+    def test_show_switch(self, switchinit):
+
+        assert json.loads(api.show_switch('sw0')) == {
+            'name': 'sw0',
+            'ports': [{'label': '3'}]
+        }
+
+        api.switch_register_port('sw0', 'test_port')
+        assert json.loads(api.show_switch('sw0')) == {
+            'name': 'sw0',
+            'ports': [{'label': '3'},
+                      {'label': 'test_port'}]
```

Comment:
I agree with that, however I am not that concerned. The test cases are fixed and we will obey the orders, besides I don't think there is a good solution to sort the ports without breaking the json object.